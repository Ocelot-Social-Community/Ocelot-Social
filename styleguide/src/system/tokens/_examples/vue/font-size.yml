#
# FONT-SIZE TOKENS
# Use these tokens to set a font-size.
#

props:
  - name: font-size-xxxx-large
    value: "3rem"
  - name: font-size-xxx-large
    value: "2.5rem"
  - name: font-size-xx-large
    value: "2rem"
  - name: font-size-x-large
    value: "1.5rem"
  - name: font-size-large
    value: "1.25rem"
  - name: font-size-base
    value: "1rem"
  - name: font-size-body
    value: "15px"
  - name: font-size-small
    value: "0.8rem"
  - name: font-size-x-small
    value: "0.7rem"
  - name: font-size-xx-small
    value: "0.6rem"
global:
  type: number
  category: font-size
