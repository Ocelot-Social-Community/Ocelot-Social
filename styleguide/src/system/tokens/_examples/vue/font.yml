#
# FONT TOKENS
# Use these tokens to set font-weight and font-family.
#

props:
  - name: font-family-heading
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-text
    value: "'Source Sans Pro', sans-serif"
    category: "font-family"
  - name: font-family-code
    value: "inconsolata, monospace"
    category: "font-family"

  - name: font-weight-regular
    value: "normal"
    category: "font-weight"
  - name: font-weight-bold
    value: "600"
    category: "font-weight"
global:
  type: ...
  category: font
