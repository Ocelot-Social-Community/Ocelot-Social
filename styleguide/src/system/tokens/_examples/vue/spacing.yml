#
# SPACING TOKENS
# Use these tokens to set padding, margin and position coordinates.
#

aliases:
  xxx-large:
    value: &xxx-large 120
  xx-large:
    value: &xx-large 60
  x-large:
    value: &x-large 40
  large:
    value: &large 28
  base:
    value: &base 20
  small:
    value: &small 12
  x-small:
    value: &x-small 6
  xx-small:
    value: &xx-small 4
  xxx-small:
    value: &xxx-small 2

props:
  - name: xxx-large
    value: *xxx-large
    category: space-size
  - name: xx-large
    value: *xx-large
    category: space-size
  - name: x-large
    value: *x-large
    category: space-size
  - name: large
    value: *large
    category: space-size
  - name: base
    value: *base
    category: space-size
  - name: small
    value: *small
    category: space-size
  - name: x-small
    value: *x-small
    category: space-size
  - name: xx-small
    value: *xx-small
    category: space-size
  - name: xxx-small
    value: *xxx-small
    category: space-size

  - name: space-xxx-large
    value: "{!xxx-large}px"
  - name: space-xx-large
    value: "{!xx-large}px"
  - name: space-x-large
    value: "{!x-large}px"
  - name: space-large
    value: "{!large}px"
  - name: space-base
    value: "{!base}px"
  - name: space-small
    value: "{!small}px"
  - name: space-x-small
    value: "{!x-small}px"
  - name: space-xx-small
    value: "{!xx-small}px"
  - name: space-xxx-small
    value: "{!xxx-small}px"

global:
  type: number
  category: space
