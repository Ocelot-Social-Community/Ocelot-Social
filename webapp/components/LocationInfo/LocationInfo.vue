<template>
  <div :class="`location-info size-${size}`">
    <div class="location">
      <base-icon name="map-marker" />
      {{ locationData.name }}
    </div>
    <div v-if="locationData.distanceToMe !== null" class="distance">
      {{ $t('location.distance', { distance: locationData.distanceToMe }) }}
    </div>
  </div>
</template>

<script>
export default {
  name: 'LocationInfo',
  props: {
    locationData: { type: Object, default: null },
    size: {
      type: String,
      default: 'base',
      validator: (value) => {
        return value.match(/(small|base)/)
      },
    },
  },
}
</script>

<style scoped>
.location-info {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  .location {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

.size-base {
  > .distance {
    margin-top: 8px;
  }
}

.size-small {
  font-size: 0.8rem;
  color: #70677e;
  margin-bottom: 12px;

  > .distance {
    margin-top: 2px;
  }
}
</style>
