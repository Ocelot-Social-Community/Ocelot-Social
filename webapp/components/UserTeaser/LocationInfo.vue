<template>
  <div class="location-info">
    <div class="location">
      <base-icon name="map-marker" />
      {{ location.name }}
    </div>
    <div v-if="distance" class="distance">{{ distance }}</div>
  </div>
</template>

<script>
export default {
  name: 'LocationInfo',
  props: {
    location: { type: Object, default: null },
  },
  computed: {
    distance() {
      return !this.location.distance
        ? null
        : this.$t('location.distance', { distance: this.location.distance })
    },
  },
}
</script>

<style scoped>
.location-info {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  .location {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .distance {
    margin-top: 8px;
  }
}
</style>
