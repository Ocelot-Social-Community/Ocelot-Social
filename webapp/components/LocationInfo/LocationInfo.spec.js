import { render } from '@testing-library/vue'
import LocationInfo from './LocationInfo.vue'

const localVue = global.localVue

describe('LocationInfo', () => {
  const Wrapper = ({ withDistance }) => {
    return render(LocationInfo, {
      localVue,
      propsData: {
        locationData: {
          name: 'Paris',
          distanceToMe: withDistance ? 100 : null,
        },
      },
      mocks: {
        $t: jest.fn((t) => t),
      },
    })
  }

  describe('distance', () => {
    it('renders with distance', () => {
      const wrapper = Wrapper({ withDistance: true })
      expect(wrapper.container).toMatchSnapshot()
    })

    it('renders without distance', () => {
      const wrapper = Wrapper({ withDistance: false })
      expect(wrapper.container).toMatchSnapshot()
    })
  })

  describe('size', () => {
    it('renders in base size', () => {
      const wrapper = Wrapper({ size: 'base' })
      expect(wrapper.container).toMatchSnapshot()
    })

    it('renders in small size', () => {
      const wrapper = Wrapper({ size: 'small' })
      expect(wrapper.container).toMatchSnapshot()
    })
  })
})
