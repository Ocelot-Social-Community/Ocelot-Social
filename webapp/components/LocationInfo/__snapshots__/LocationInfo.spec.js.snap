// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationInfo renders with distance 1`] = `
<div>
  <div
    class="location-info size-base"
  >
    <div
      class="location"
    >
      <span
        class="base-icon"
      >
        <!---->
      </span>
      
    Paris
  
    </div>
     
    <div
      class="distance"
    >
      location.distance
    </div>
  </div>
</div>
`;

exports[`LocationInfo renders without distance 1`] = `
<div>
  <div
    class="location-info size-base"
  >
    <div
      class="location"
    >
      <span
        class="base-icon"
      >
        <!---->
      </span>
      
    Paris
  
    </div>
     
    <!---->
  </div>
</div>
`;
